pkg/reconciler/configuration/configuration_gcv2_test.go

```diff
@@ -0,0 +1,576 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configuration
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	// Inject the fake informers we need.
+	_ "knative.dev/serving/pkg/client/injection/informers/serving/v1/configuration/fake"
+	_ "knative.dev/serving/pkg/client/injection/informers/serving/v1/revision/fake"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/clock"
+	clientgotesting "k8s.io/client-go/testing"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/ptr"
+	cfgMap "knative.dev/serving/pkg/apis/config"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+	servingclient "knative.dev/serving/pkg/client/injection/client/fake"
+	configreconciler "knative.dev/serving/pkg/client/injection/reconciler/serving/v1/configuration"
+	"knative.dev/serving/pkg/reconciler/configuration/resources"
+
+	. "knative.dev/pkg/reconciler/testing"
+	. "knative.dev/serving/pkg/reconciler/testing/v1"
+	. "knative.dev/serving/pkg/testing/v1"
+)
+
+var revisionSpec = v1.RevisionSpec{
+	PodSpec: corev1.PodSpec{
+		Containers: []corev1.Container{{
+			Image: "busybox",
+		}},
+	},
+	TimeoutSeconds: ptr.Int64(60),
+}
+
+var testCtx context.Context
+var testClock clock.Clock
+
+// This is heavily based on the way the OpenShift Ingress controller tests its reconciliation method.
+func TestReconcileNewGCEnabled(t *testing.T) {
+	retryAttempted := false
+	now := time.Now()
+	testClock = clock.NewFakeClock(now)
+
+	c := &cfgMap.Config{
+		Features: &cfgMap.Features{
+			ResponsiveRevisionGC: cfgMap.Enabled,
+		},
+	}
+	testCtx = cfgMap.ToContext(context.Background(), c)
+
+	table := TableTest{{
+		Name: "bad workqueue key",
+		Key:  "too/many/parts",
```

Comment:
Oh I meant just create a new test covering _only_ your new scenarios 

ie. when the config reconciler creates a new revision it has the pending route state. 

All these other tests don't need to be duplicated.

Reply:
Sure. Since the rev() helper is in both the setup and wantCreate, it's a little nonobvious what this is testing (it's not really testing a new scenario). The more obvious unit test for the annotation is in revision_test, but here we want to ensure there are not new side effects after enabling it.